tests/unittests/test_dispatcher.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import os
+from azure_functions_worker import protos
+from azure_functions_worker import testutils
+from azure_functions_worker.constants import PYTHON_THREADPOOL_THREAD_COUNT
+
+
+class TestDispatcher(testutils.AsyncTestCase):
+    dispatcher_funcs_dir = testutils.UNIT_TESTS_FOLDER / 'dispatcher_functions'
+
+    def setUp(self):
+        self._pre_env = dict(os.environ)
+
+    def tearDown(self):
+        os.environ.clear()
+        os.environ.update(self._pre_env)
+
+    async def test_dispatcher_sync_threadpool_default_worker(self):
+        '''Test if the sync threadpool has maximum worker count set to 1
+        by default
+        '''
+        ctrl = testutils.start_mockhost(script_root=self.dispatcher_funcs_dir)
+
+        async with ctrl as host:
+            # Ensure the function can be properly loaded
+            func_id, load_r = await host.load_function('show_context')
+            self.assertEqual(load_r.response.function_id, func_id)
+            self.assertEqual(load_r.response.result.status,
+                             protos.StatusResult.Success)
+
+            # Ensure the function can be properly invoked
+            invoke_id, call_r = await host.invoke_function(
+                'show_context', [
+                    protos.ParameterBinding(
+                        name='req',
+                        data=protos.TypedData(
+                            http=protos.RpcHttp(
+                                method='GET'
+                            )
+                        )
+                    )
+                ])
+            self.assertIsNotNone(invoke_id)
+            self.assertEqual(call_r.response.result.status,
+                             protos.StatusResult.Success)
+
+            # Ensure the dispatcher sync threadpool count is set to 1
+            self.assertEqual(ctrl._worker._sync_tp_max_workers, 1)
+
+    async def test_dispatcher_sync_threadpool_set_worker(self):
```

Comment:
You might also want to add a `check_logs_*` test as well to see if the logline came in from the host.

Reply:
The testutils.AsyncTestCase case does not have check_logs_ utils in it. I will create another test class to validate the error logs.

Reply:
The check_logs_* check does not work here since the threadpool initialization starts before the function invocation.
The check_log host_out only records the invocation context because the `self.host_stdout.seek(last_pos)` the cursor keeps updating from the last test. https://github.com/Azure/azure-functions-python-worker/blob/dev/azure_functions_worker/testutils.py#L225.

Thus, I think the best I can do is to test the logging with a mock logger.

Reply:
ah - sounds good. Thanks for the info.